Src/ILGPU/Runtime/Cuda/CudaNativeMethods.cs

```diff
@@ -284,7 +285,69 @@ public enum MemHostRegisterFlags
         CU_MEMHOSTREGISTER_READ_ONLY = 8,
     }
 
+    /// <summary>
+    /// Controls the behaviour of <see cref="CudaAPI.cuIpcOpenMemHandle">IpcOpenMemHandle</see>
+    /// </summary>
+    [Flags]
+    public enum CudaIpcMemFlags
+    {
+        /// <summary>
+        /// No flags.
+        /// </summary>
+        None,
+        /// <summary>
+        /// Enables peer access lazily.
+        /// </summary>
+        /// <remarks>
+        /// From <a href="https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g60f28a5142ee7ae0336dfa83fd54e006">
+        /// CUDA Runtime Types
+        /// </a>
+        /// </remarks>
+        LazyEnablePeerAccess = 1 << 0
+    }
     #endregion
+
+    /// <summary>
+    /// Represents a CUDA IPC memory handle.
+    /// </summary>
+    /// <remarks>This represents the struct
+    /// <a href="https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaIpcMemHandle__t.html#structcudaIpcMemHandle__t">cudaIpcMemHandle__t</a>.
+    /// </remarks>
+    public struct CudaIpcMemHandle
```

Comment:
QUESTION: Is there a reason for having a struct with another struct (i.e. `Handle` within `CudaIpcMemHandle`)?

Why not just declare a single struct?

Reply:
Originally it was that way to mirror `cudaIpcMemHandle__t` having the array as a field. 
Having `[InlineArray(64)]` directly on `CudaIpcMemHandle` means it straight up **is** the array.
Also honestly speaking, I didn't think to test it that way, asg etting arrays to work with `LibraryImport` instead of `DllImport` was a already a bit of a hassle.
That being said, I tested it that way and it actually worked. Since we don't have any other fields in the struct, we should actually be fine this way.
To ease the usage, since InlineArrays can't be used directly as arrays, I added explicit conversions to and from `byte[]`. 
I thought about making them implicit, but decided against it because they should not throw exceptions and generally carry the assumption of being allocation free.